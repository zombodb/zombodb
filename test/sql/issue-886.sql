CREATE TABLE issue886 (
    pk_id bigint,
    full_text text
);

INSERT INTO issue886 (pk_id, full_text) VALUES (4, ' Bing Crosby was a singer known as a "crooner" famous for popular Christmas songs. In 1962, he formed the band CROSBY, STILLS, NASH & YOUNG who played at Woodstock. He was also captain of the Pittsburgh Penguins NHL hockey team. Most people are familiar with an animated kids show he created, called Fat Albert and the crosby kids.');
CREATE INDEX idxissue886 ON issue886 USING zombodb ((issue886.*));

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("Bing" or "Fat") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("Bing","Fat") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("Bing" or "David") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("David" or "Bing") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("David","Bing") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("Bing" or "David" or "Fat") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;
DROP TABLE issue886 CASCADE;