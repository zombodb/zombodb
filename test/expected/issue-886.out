CREATE TABLE issue886 (
    pk_id bigint,
    full_text text
);
INSERT INTO issue886 (pk_id, full_text) VALUES (4, ' Bing Crosby was a singer known as a "crooner" famous for popular Christmas songs. In 1962, he formed the band CROSBY, STILLS, NASH & YOUNG who played at Woodstock. He was also captain of the Pittsburgh Penguins NHL hockey team. Most people are familiar with an animated kids show he created, called Fat Albert and the crosby kids.');
CREATE INDEX idxissue886 ON issue886 USING zombodb ((issue886.*));
WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("Bing" or "Fat") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;
 field_name | array_index |  term  |    type    | position | start_offset | end_offset |              query_clause               | primary_key 
------------+-------------+--------+------------+----------+--------------+------------+-----------------------------------------+-------------
 full_text  |           0 | bing   | <ALPHANUM> |        1 |            1 |          5 | full_text:(("bing","fat") W/2 "crosby") |           4
 full_text  |           0 | crosby | <ALPHANUM> |        2 |            6 |         12 | full_text:(("bing","fat") W/2 "crosby") |           4
(2 rows)

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("Bing","Fat") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;
 field_name | array_index |  term  |    type    | position | start_offset | end_offset |              query_clause               | primary_key 
------------+-------------+--------+------------+----------+--------------+------------+-----------------------------------------+-------------
 full_text  |           0 | bing   | <ALPHANUM> |        1 |            1 |          5 | full_text:(("bing","fat") W/2 "crosby") |           4
 full_text  |           0 | crosby | <ALPHANUM> |        2 |            6 |         12 | full_text:(("bing","fat") W/2 "crosby") |           4
(2 rows)

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("Bing" or "David") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;
 field_name | array_index |  term  |    type    | position | start_offset | end_offset |               query_clause                | primary_key 
------------+-------------+--------+------------+----------+--------------+------------+-------------------------------------------+-------------
 full_text  |           0 | bing   | <ALPHANUM> |        1 |            1 |          5 | full_text:(("bing","david") W/2 "crosby") |           4
 full_text  |           0 | crosby | <ALPHANUM> |        2 |            6 |         12 | full_text:(("bing","david") W/2 "crosby") |           4
(2 rows)

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("David" or "Bing") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;
 field_name | array_index |  term  |    type    | position | start_offset | end_offset |               query_clause                | primary_key 
------------+-------------+--------+------------+----------+--------------+------------+-------------------------------------------+-------------
 full_text  |           0 | bing   | <ALPHANUM> |        1 |            1 |          5 | full_text:(("david","bing") W/2 "crosby") |           4
 full_text  |           0 | crosby | <ALPHANUM> |        2 |            6 |         12 | full_text:(("david","bing") W/2 "crosby") |           4
(2 rows)

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("David","Bing") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;
 field_name | array_index |  term  |    type    | position | start_offset | end_offset |               query_clause                | primary_key 
------------+-------------+--------+------------+----------+--------------+------------+-------------------------------------------+-------------
 full_text  |           0 | bing   | <ALPHANUM> |        1 |            1 |          5 | full_text:(("david","bing") W/2 "crosby") |           4
 full_text  |           0 | crosby | <ALPHANUM> |        2 |            6 |         12 | full_text:(("david","bing") W/2 "crosby") |           4
(2 rows)

WITH matches as MATERIALIZED (
    SELECT * FROM issue886 WHERE pk_id = 4),
     highlights AS MATERIALIZED (
         SELECT (
                    zdb.highlight_document('issue886'::regclass,
                                           json_build_object('full_text',full_text),
                                           'full_text:(("Bing" or "David" or "Fat") w/2 "crosby")'::TEXT)
                    ).*,
                pk_id AS primary_key FROM matches
     )
SELECT * FROM highlights;
 field_name | array_index |  term  |    type    | position | start_offset | end_offset |                  query_clause                   | primary_key 
------------+-------------+--------+------------+----------+--------------+------------+-------------------------------------------------+-------------
 full_text  |           0 | bing   | <ALPHANUM> |        1 |            1 |          5 | full_text:(("bing","david","fat") W/2 "crosby") |           4
 full_text  |           0 | crosby | <ALPHANUM> |        2 |            6 |         12 | full_text:(("bing","david","fat") W/2 "crosby") |           4
(2 rows)

DROP TABLE issue886 CASCADE;
